### ISSUE TITLE
typesccript reports errors with FormData since 13.0.10

### ISSUE THREAD

**Issue by jeremyriverain** [acacode/swagger-typescript-api#814]
I'm experiencing TypeScript errors that come from the generated `http-client.ts`.

Here are the errors:

![Capture d’écran du 2024-07-01 12-26-26](https://github.com/acacode/swagger-typescript-api/assets/23351988/23253b60-0135-4f60-a292-174839cd67f0)

My build passes with version `13.0.10` of swagger-typescript-api but not with version `13.0.11`.

By comparing the 2 generated files, here are the differences I obtain:

![Capture d’écran du 2024-07-01 12-23-06](https://github.com/acacode/swagger-typescript-api/assets/23351988/0de8c412-1d8c-40f3-a7fd-fd679b62363b)

The file on the left is the one generated by swagger-typescript-api version `13.0.11`. It has TypeScript errors.

The file on the right does not have TypeScript error.

My project uses version 5.2 of typescript.

I think the bug is introduced by this commit: https://github.com/acacode/swagger-typescript-api/commit/0d80f8e6ea5a2420745841f8b9eb95faae0b4ffc

**Comment by smorimoto** [comment]
Provide specific reproducible example.

**Comment by am-burban** [comment]
Hey,
I'm wondering what the intention of this change is. If I need to pass in an instance of `FormData` as the request body, why do I need the formatter then? It just recreates the `FormData` instance again (and possibly loses information, because it e.g. does not support he third parameter of `FormData.prototype.append`).
For me the main use case of this formatter was, that I can pass in a plain object, that is transformed to a `FormData` instance. This is broken (by-design?) now, but imho shouldn't have happened in a patch release.

**Comment by philippone** [comment]
Hello @jeremyriverain ,

I guess you have to add `"dom.iterable"` into your tsconfig:

`
"compilerOptions": {
        "lib": ["esNext", "dom", "dom.iterable"],
`

**Comment by svarion1** [comment]
Hi I second this bug. After the update the autogenerated code is generating in the http-client new code with `Array.from` instead of `Object.keys`. 
This then brings the same error as Jeremy.

I had already in my compilerOptions the `"dom.iterable"`

**Comment by payammeyer** [comment]
Same issue occurs with `13.0.12`.

**Comment by motionrus** [comment]
I renamed the file from `api.ts` to `api.d.ts` and the error disappeared

**Comment by AlexanderShabalinMaersk** [comment]
I am wondering - will it just stay as it is? It is a breaking change in a patch version, right?

**Comment by benvd** [comment]
I've created a reproducible example here: [benvd/swagger-typescript-api-814](https://github.com/benvd/swagger-typescript-api-814).

**Comment by smorimoto** [comment]
@benvd Indeed! It was definitely a bug! I will revert the PR and release it as 13.0.16.

**Comment by smorimoto** [comment]
https://github.com/acacode/swagger-typescript-api/releases/tag/13.0.16
